### PR TITLE
mobile account nav 

### LINKED ISSUE #140: Mobile nav for account settings

### LINKED ISSUE #165: Account Settings not centered like other screens
Go to the account settings. The main content is off to the right, not centered like on other pages

https://user-images.githubusercontent.com/444356/184652545-05753096-89da-40eb-ab23-7621b0240800.mov

### LINKED ISSUE #166: There's no header or footer on the Manage Profile screen
Go to Manage Profile in the menu
The page has no header or footer. I expect it to have both.

Screen Shot 2022-08-15 at 10 19 25 AM

### LINKED ISSUE #182: Create account with email/password needs error feedback for incomplete rules
Go to sign up with email. Enter email. Enter  password. Forget to comply to one or more of the password rule and press continue. Missing error feedback for rules and password input.

Screen Shot 2022-08-23 at 10 28 30 AM
 
Expected behavior: error feedback

Screen Shot 2022-08-23 at 10 28 50 AM

### LINKED ISSUE #183: Add eye icon to password fields in create account and sign in cards
When typing in password, the eye icon should appear to allow user to toggle between showing and hiding password

### PR DESCRIPTION
**Pull request by Carla-Moz**
**This PR adds the mobile navigation bar to the account page.**

Screen Shot 2022-08-26 at 1 41 54 PM

Screen Shot 2022-08-26 at 1 42 14 PM

Screen Shot 2022-08-26 at 2 10 06 PM

**Aligns all the view (Studies, Profile, and account):**
Screen Shot 2022-08-26 at 9 25 54 PM

Screen Shot 2022-08-26 at 9 26 03 PM

Screen Shot 2022-08-26 at 9 26 25 PM

**And adds the nav and footer to the profile page:**
Screen Shot 2022-08-26 at 9 31 05 PM

**Adds error feedback to field (red with warning icon) when user breaks the password rules during email signup:**
Screen Shot 2022-08-29 at 10 24 20 AM

**And finally, adds eyes icons to signup password forms to users can show and hide password as they type:**
Screen Shot 2022-08-29 at 11 07 58 AM

Screen Shot 2022-08-29 at 11 08 28 AM

closes #140
closes #165 
closes #166 
closes #182 
closes #183

### CODE REVIEW

**Comment by jepstein** [comment]
tested and all updates are working as expected

File: web-platform/website/src/components/navigation-bar/NavigationBar.tsx [hunk 1]
```diff
@@ -126,4 +125,14 @@ const styles = {
       },
     }
   ),
+  logoCol: style(
+    {
+      marginRight: `0`,
```
Comment (thomik-corp) [inline web-platform/website/src/components/navigation-bar/NavigationBar.tsx:130]: You can just tag me-0 on line 30. If we need this style we can just use numeric for marginRight as in marginRight: 0

File: web-platform/website/src/components/pages/account-settings/AccountSettingsMobileNavigationBar.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      {strings.sections.map((section) => {
+        if (section.links) {
+          return <AccordionSection {...section} key={section.text} />;
+        }
+      })}
+    </Container>
+  );
+}
+
+function AccordionSection(section: {
```
Comment (thomik-corp) [inline web-platform/website/src/components/pages/account-settings/AccountSettingsMobileNavigationBar.tsx:39]: Suggest we rename this to AccountSettingsSection

File: web-platform/website/src/components/pages/account-settings/AccountSettingsNavigationBarWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { AccountSettingsMobileNavigationBar } from "./AccountSettingsMobileNavigationBar";
```
Comment (thomik-corp) [inline web-platform/website/src/components/pages/account-settings/AccountSettingsNavigationBarWrapper.tsx:1]: This seems like an overkill for a class esp we are using it just once. Should we kill it and inline calls from the caller?

File: web-platform/website/src/components/pages/account-settings/AccountSettingsPageContent.tsx [hunk 4]
```diff
@@ -1,16 +1,16 @@
 import { Col, Container, Row } from "reactstrap";
 import { style } from "typestyle";
 
-import { Spacing } from "../../../styles";
-import { ContainerStyles } from "../../../styles/ContainerStyles";
+import { ScreenSize, Spacing, createResponsiveStyle } from "../../../styles";
```
Comment (thomik-corp) [inline web-platform/website/src/components/pages/account-settings/AccountSettingsPageContent.tsx:4]: I suspect if you run npm run lint or npm run build it should lint the PR. I installed prettier in VSCode and using config from tools/typescript/.prettierrc to format as I save.

File: web-platform/website/src/styles/Columns.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+
+import { style } from "typestyle";
+import { ScreenSize } from "./ScreenSize";
+import { Spacing } from "./Spacing";
+import { createResponsiveStyle } from "./ResponsiveStyle";
+
+export const ColumnStyles = {
```
Comment (thomik-corp) [inline web-platform/website/src/styles/Columns.ts:7]: Thinking if we should move it to account-settings and rename the ColumnStyles to AccountSettingColumnStyles.

**Comment by thomik-corp** [comment]
Looks great functionally. I have provided some suggestions and linting request.

File: web-platform/website/src/components/pages/account-settings/AccountSettingsNavigationBarWrapper.tsx (hunk 3 above)
Comment (Carla-Moz) [reply web-platform/website/src/components/pages/account-settings/AccountSettingsNavigationBarWrapper.tsx:1]: True lol

File: web-platform/website/src/styles/Columns.ts (hunk 5 above)
Comment (Carla-Moz) [reply web-platform/website/src/styles/Columns.ts:7]: Hmm, I'm not sure. We might need unique column styles for the other pages depending on what Casey creates. Let's keep it for now and if it's unnecessary, we'll move and rename it. We could also create a folder called pages in the styles dir for the pages.

**Pull request merged**